### PR TITLE
Add Yarn and Lerna to project, remove CI

### PR DESCRIPTION
**Pull request by contolini**
Adds Lerna for monorepo management and Yarn for it's neato workspaces and caching features.

Check out the [contributing docs](https://github.com/cfpb/capital-framework/blob/0e95345db34ad8e9a6349af8468ca4101b311292/CONTRIBUTING.md#changing-the-codebase) to see how these changes affect CF development. Spoiler alert: The only thing you'll have to do differently is remember to use `yarn` instead of `npm`.

### Notable Changes

1. The `src/` directory in the project's root has been renamed to `packages/`.
1. All the Travis CI stuff has been removed. We'll manually publish components (which is [easy with Lerna](https://github.com/cfpb/capital-framework/blob/7f1e16add97dd5abd91f6d9b72feba4ac149b057/CONTRIBUTING.md#release-management)) for the time being and we can move the task back into Travis if/when we're ready.
1. No more npm or `package-lock.json`. Node modules are cached in the git repo.
1. No more scary scripts for publishing/linking. Yarn and Lerna do most of it natively.

### Testing

1. Pull down this branch (might take a minute with all the cached packages).
1. `rm -rf node_modules/` to ensure you're starting with a clean slate.
1. `yarn` to install (no more `npm install`)
1. `yarn test` to confirm the tests pass as normal.
1. `yarn run cf-link` to link all the components
1. Edit `/packages/cf-core/src/cf-vars.less` and set the text color on line 13 to something silly like `red`.
1. `cd ~/wherever/cfgov-refresh`
1. `yarn link cf-core`
1. `gulp clean && gulp build`
1. Start up cfgov-refresh and open http://0.0.0.0:8000/. Text should now be red, indicating that it's correctly picking up your local version of `cf-core`.

### Unanswered questions

1. It's currently configured to use [fixed](https://github.com/lerna/lerna#how-it-works) versioning. Should we use independent versioning instead? We've been split on this. Let's keep discussing. **Update:** We discussed this and are happy with fixed versioning.

### CODE REVIEW

**Comment by anselmbradford** [comment]
any testing instructions?

**Comment by contolini** [comment]
Yeah, I'll paste some here after I do a sanity check on the FEWD bi-weekly.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,26 +25,25 @@ guidance below.
 ## Changing the codebase
 
 We are no longer forking and pulling. Instead, we are working off feature branches.
-After you've edited a component, open a PR to merge your feature branch back into `canary`.
+After you've edited a component, open a PR to merge your feature branch back into `master`.
 
 For example, if you wanted to change `cf-buttons`, here's what you'd do:
```
Comment (cfarm) [inline CONTRIBUTING.md]: I never reference this section so I never noticed this (I always use the "Testing in the documentation site" instructions) but I would just make sure this sentence describes the fact that the steps are changing cf-buttons in Capital Framework, while compiling and viewing changes in the cfgov-refresh app. That's sort of implied but not made clear.

**Comment by contolini** [comment]
Hey look, I finally rebased this branch and added testing instructions! Nice job @contolini!

After this is merged I'd like to do a few test patch releases to kick the tires and confirm `lerna-changelog` works how we expect. I'll record the process and add the video to our docs.

**Comment by contolini** [comment]
Snyk can be ignored because it's looking for the manifests in `/src/` and not `/packages/`. Once master is updated we'll need to delete and re-add the repo to snyk to have it pick up the new paths.

File: scripts/generate-changelog.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+# Generate a changelog for all PRs after the Lerna migration (git tag v7.7.0)
+# See docs at https://github.com/lerna/lerna-changelog
+lerna-changelog --from v7.7.0 > CHANGELOG.md
+
+# Add the pre-Lerna changelog to the bottom of the generated one.
+cat ./scripts/templates/CHANGELOG.legacy.md >> ./CHANGELOG.md
```
Comment (contolini) [inline scripts/generate-changelog.sh]: I put the current changelog into a file called `CHANGELOG.legacy.md` and it gets added to the bottom of the fresh changelog that will be generated by `lerna-changelog` going forward.

**Comment by Scotchester** [comment]
@contolini Can you tell me how to see what the generated changelog will look like?

**Comment by Scotchester** [comment]
There's a lot of repeated content in each package's meta files (`CONTRIBUTING`, `LICENSE`, `README`, `TERMS`, etc.). Are those generated from a single source and then checked in in this PR? If not, can they be?

**Comment by contolini** [comment]
> Can you tell me how to see what the generated changelog will look like?

Unfortunately `lerna-changelog` uses GH PR titles so I haven't been able to test it on this repo until we merge this PR. You can see the format over in [bapital's changelog](https://github.com/cfpb/bapital-framework/blob/master/CHANGELOG.md).

> There's a lot of repeated content in each package's meta files (`CONTRIBUTING`, `LICENSE`, `README`, `TERMS`, etc.). Are those generated from a single source and then checked in in this PR? If not, can they be?

This was intentional because I wanted to remove all "magic" from the publishing process. Each package exists in this repo exactly as it does in npm (and thus other project's `node_modules/` directories). We rarely, if ever, update the above files so I didn't see it necessary to automate their creation. We can automate it later if it proves to be helpful. We'll likely want to implement other automation steps (like Travis publishing) down the road.

**Pull request merged**